src/Agoda.Analyzers.Test/AgodaCustom/AG0039UnitTests.cs

```diff
@@ -0,0 +1,39 @@
+using System.Threading.Tasks;
+using Agoda.Analyzers.AgodaCustom;
+using Agoda.Analyzers.Test.Helpers;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Framework;
+
+namespace Agoda.Analyzers.Test.AgodaCustom;
+
+class AG0039UnitTests : DiagnosticVerifier
+{
+    protected override DiagnosticAnalyzer DiagnosticAnalyzer => new AG0039UndocumentedMemberAnalyzer();
+
+    protected override string DiagnosticId => AG0039UndocumentedMemberAnalyzer.DIAGNOSTIC_ID;
+
+    [Test]
+    public async Task AG0037_WithRegion_ShowsWarning()
+    {
+        var code = @"
+				public class NotDoc
```

Comment:
Can we have a negative case for checking that the rule does not report compliant code?